pkgs/tools/misc/mstflint/default.nix

```diff
@@ -4,23 +4,85 @@
 , libibmad
 , openssl
 , zlib
+, xz
+, expat
+, boost
+, curl
+, pkg-config
+, autoreconfHook
+, libxml2
+, pciutils
+, busybox
+, python3
+# use this to shrink the package's footprint if necessary (e.g. for hardened appliances)
+, onlyFirmwareUpdater ? false
 }:
 
 stdenv.mkDerivation rec {
   pname = "mstflint";
-  version = "4.17.0-1";
+  version = "4.25.0-1";
 
   src = fetchurl {
     url = "https://github.com/Mellanox/mstflint/releases/download/v${version}/mstflint-${version}.tar.gz";
-    sha256 = "030vpiv44sxmjf0dng91ziq1cggwj33yp0l4xc6cdhnrv2prjs7y";
+    hash = "sha256-nYGiWfr8a3q3+bGUb1ovLrAS8/LnEJf+4inIEllW95s=";
   };
 
+  nativeBuildInputs = [
+    autoreconfHook
+    pkg-config
+  ];
+
   buildInputs = [
     libibmad
-    openssl
     zlib
+    libxml2
+    openssl
+  ] ++ lib.optionals (!onlyFirmwareUpdater) [
+    boost
+    curl
+    expat
+    xz
+    python3
+  ];
+
+  # tracked upstream at
+  #  - https://github.com/Mellanox/mstflint/pull/855
+  #  - https://github.com/Mellanox/mstflint/issues/856
+  # remove if merged
```

Comment:
The patches seem to be merged now. Can you ask for a new release? If not, I'd rather switch to an unstable version than replicate the patches here.

Reply:
@nikstur I'm currently testing with the upstream master_devel Branch. I'll notify you, when this PR is ready again.

Reply:
4.26 was released and I updated + rebased the PR accordingly.